src/shared/qrcode-util.c

```diff
@@ -116,4 +185,49 @@ int print_qrcode(FILE *out, const char *header, const char *string) {
         sym_QRcode_free(qr);
         return 0;
 }
+
+int print_positioned_qrcode(FILE *out, const char *header, const char *string, unsigned row, unsigned column, unsigned tty_width, unsigned tty_height) {
+        QRcode* qr;
+        int r, fd;
+        unsigned qr_code_width, qr_code_height;
+
+        fd = fileno(out);
+        if (fd < 0)
+                return log_warning_errno(fd, "unable to get file descriptor from the file stream: %m");
+
+        /* If this is not an UTF-8 system or ANSI colors aren't supported/disabled don't print any QR
+         * codes */
+        if (!is_locale_utf8() || !colors_enabled())
+                return -EOPNOTSUPP;
+
+        r = dlopen_qrencode();
+        if (r < 0)
+                return r;
+
+        qr = sym_QRcode_encodeString(string, 0, QR_ECLEVEL_L, QR_MODE_8, 1);
+        if (!qr)
+                return -ENOMEM;
+
+        qr_code_width = qr_code_height = qr->width + 8;
+
+        if (header) {
+                set_terminal_cursor_position(fd, row - 2, tty_width - qr_code_width - 2);
+                fprintf(out, "%s:\n\n", header);
+        }
+
+        if (column + qr_code_width > tty_width)
+                column = tty_width - qr_code_width;
+
+        /*Terminal characters are twice as high as they are wide so it's qr_code_height / 2,
+         *  our QR code prints an extra new line, so we have -1 as well */
+        if (row + qr_code_height > tty_height)
+                row = tty_height - (qr_code_height / 2 ) - 1;
```

Comment:
@bluca, I found the explanation for the column issue, and I've added it in the comment here.